### DISCUSSION TITLE
Work flow with raw data

### DISCUSSION THREAD

**Discussion by AlvicePC** [PeterMakus/SeisMIC#68] [Q&A] [answered]
Hi, i tried to make the work flow with some mseed archives that i have in my pc, but in the examples i need to use the class fdsn and i trace data using FS_Client.

**import os
from obspy import UTCDateTime
from datetime import datetime
from seismic.trace_data.waveform import FS_Client
from seismic.trace_data.waveform import read_from_filesystem
try: ipynb_path
except NameError: ipynb_path = os.getcwd()

os.chdir(ipynb_path)

root = os.path.join(ipynb_path, 'data')
os.makedirs(root, exist_ok=True)

start = datetime(2021, 1, 1)
end = datetime(2021, 2, 1)
network = 'JA'
station = 'VCCA'

sc = FS_Client(fs=['/home/alvaro/project/data/mseed',
          '%Y', '%NET', '%STA', ['%CHA', '.D'],
          ['%NET','.','%STA','.','.','%CHA','.D.','%Y','.','%j']])

sc.get_waveforms(network=network, station=station, location='*', channel='HHZ', starttime=start, endtime=end)**

and it backs
1 Trace(s) in Stream:
**JA.VCCA..HHZ | 2021-01-01T00:00:00.000000Z - 2021-01-31T23:59:59.990000Z | 100.0 Hz, 267840000 samples**

in the next step i edit the correlate.py to this

import os
os.environ['OPENBLAS_NUM_THREADS'] = '1'

from time import time
from seismic.trace_data.waveform import FS_Client
from seismic.correlate.correlate import Correlator

params = 'params.yaml'

root = '/home/alvaro/project/data'
sc = FS_Client(fs=['/home/alvaro/Downloads/JA_2021/JA/data/mseed',
          '%Y', '%NET', '%STA', ['%CHA', '.D'],
          ['%NET','.','%STA','.','.','%CHA','.D.','%Y','.','%j']])

c = Correlator(sc, options=params)
print('Correlator initiated')
x = time()
st = c.pxcorr()
print('Correlation finished after', time()-x, 'seconds')

and in the terminal run 

mpirun python correlate.py

but it returns

Traceback (most recent call last):
  File "/home/alvaro/project/correlate.py", line 19, in <module>
    c = Correlator(sc, options=params)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alvaro/anaconda3/lib/python3.11/site-packages/seismic/correlate/correlate.py", line 189, in __init__
    self.store_client._translate_wildcards(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'FS_Client' object has no attribute '_translate_wildcards'
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[47922,1],0]
  Exit code:    1

**Comment by PeterMakus** [comment, marked as answer]
Hi,
The reason for this is that ``Correlator`` expects an object of type ``StoreClient`` to be fed in. You can check the [documentation](https://petermakus.github.io/SeisMIC/modules/API/API.html#seismic.trace_data.waveform.Store_Client) for details.

To use your own data, you will have to structure it in the same way as data downloaded from FDSN servers.
